src/main/scala/pyspark_util/Pickling.scala

```diff
@@ -101,6 +102,7 @@ class Pickling extends Serializable {
     Pickler.registerCustomPickler(classOf[Set.Set4[_]], ListPickler)
     Pickler.registerCustomPickler(classOf[HashSet1[_]], ListPickler)
     Pickler.registerCustomPickler(classOf[HashTrieSet[_]], ListPickler)
+    Pickler.registerCustomPickler(classOf[Map[_, _]], MapPickler)
     Pickler.registerCustomPickler(classOf[WithDefault[_, _]], MapPickler)
```

Comment:
@anguenot When I added unit tests, I found a similar issue with empty maps.

Registering a custom pickler for the `Map` type seems to fix this, but I confess I am not familiar with Scala and don't know what is going on here with the `WithDefault` and `MapN` types.

Reply:
Looking at Scala docs... it looks like these are subclasses of `Map`. Is it necessary to register the picklers for all the subtypes?
```
    Pickler.registerCustomPickler(classOf[WithDefault[_, _]], MapPickler)
    Pickler.registerCustomPickler(classOf[Map1[_, _]], MapPickler)
    Pickler.registerCustomPickler(classOf[Map2[_, _]], MapPickler)
    Pickler.registerCustomPickler(classOf[Map3[_, _]], MapPickler)
    Pickler.registerCustomPickler(classOf[Map4[_, _]], MapPickler)
```
I tried removing these lines the tests are passing for me locally, but I'm not sure that there are not potential problems not covered by the test suite.

Reply:
@ndamclean  wanna open a  new issue for that one so we can trace potential issues separately?

Reply:
I looked at the pyrolite code and concluded it's probably best to have all the subclasses registered separately for the sake of efficiency. 
```
	protected IObjectPickler getCustomPickler(Type t)
	{
		IObjectPickler pickler;
		if(customPicklers.TryGetValue(t, out pickler))
			return pickler;		// exact match
		
		// check if there's a custom pickler registered for an interface or abstract base class
		// that this object implements or inherits from.
		foreach(var x in customPicklers) {
			if(x.Key.IsAssignableFrom(t)) {
				return x.Value;
			}
		}


		return null;
	}
```